src/components/data/data-utils.js

```diff
@@ -438,14 +444,14 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
             }
           })
         ]),
-        currentFile && _.includes(_.toLower(newEntityType), entityTypes) && div({
+        ((currentFile && entityAlreadyExists) || recordAlreadyExists) && div({
           style: { ...warningBoxStyle, margin: '1rem 0 0.5rem', display: 'flex', alignItems: 'center' }
         }, [
           icon('warning-standard', { size: 19, style: { color: colors.warning(), flex: 'none', marginRight: '0.5rem', marginLeft: '-0.5rem' } }),
-          `Data with the type '${newEntityType}' already exists in this workspace. `,
+          `Data with the type '${recordOrEntity()}' already exists in this workspace. `,
```

Comment:
optional, for style: I'd inline the logic here, and delete the `recordOrEntity()` function entirely, since it's only used here.
```javascript
`Data with the type '${recordType ? recordType : newEntityType}' already exists in this workspace. `,
```

Reply:
Ah, I had been using it more widely and didn't realize it was only down to one use after refactoring.